eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/processors/ContigReplacerProcessorTest.java

```diff
@@ -38,61 +40,132 @@
+        assertEquals(GENBANK_CONTIG, processor.process(input).getContigName());
     }
 
     @Test
-    public void convertVariantFromRefSeqToSeqName() throws Exception {
-        SubSnpNoHgvs input = new SubSnpNoHgvs(SS_ID, RS_ID, REFERENCE_ALLELE, ALTERNATE_ALLELE, ASSEMBLY, "", "",
-                                              REFSEQ_1, START, null, 0L,
-                                              DbsnpVariantType.SNV, Orientation.FORWARD, Orientation.FORWARD,
-                                              Orientation.FORWARD, true, true, true, true, null, null, TAXONOMY);
-        assertEquals(SEQNAME_1, processor.process(input).getContigName());
+    public void convertContigIfChromosomeIsMissingInAssemblyReport() throws Exception {
+        SubSnpNoHgvs input = newMockSubSnpNoHgvs(MISSING_CHROMOSOME, REFSEQ_1);
+        assertEquals(GENBANK_1, processor.process(input).getContigName());
     }
 
     @Test
-    public void convertVariantFromUcscToSeqName() throws Exception {
+    public void doNotConvertToGenbankIfGenbankAndRefseqAreNotIdentical() throws Exception {
+        SubSnpNoHgvs input = newMockSubSnpNoHgvs(SEQNAME_WITHOUT_SYNONYM, REFSEQ_WITHOUT_SYNONYM);
+        assertEquals(REFSEQ_WITHOUT_SYNONYM, processor.process(input).getContigName());
+    }
+
+    @Test
+    public void doNotConvertChromosomeToGenbankIfGenbankAndRefseqAreNotIdentical() throws Exception {
+        SubSnpNoHgvs input = newMockSubSnpNoHgvs(SEQNAME_WITHOUT_SYNONYM, MISSING_CONTIG);
+        assertEquals(MISSING_CONTIG, processor.process(input).getContigName());
+    }
+
+    @Test
+    public void convertChromosomeToGenbankIfGenbankAndRefseqAreNotIdenticalAndAssemblyIsGenbank() throws Exception {
+        SubSnpNoHgvs input = newMockSubSnpNoHgvs(SEQNAME_WITHOUT_SYNONYM, MISSING_CONTIG);
+        assertEquals(GENBANK_WITHOUT_SYNONYM, genbankProcessor.process(input).getContigName());
+    }
+
+    @Test
+    public void ifConvertingFromChromosomeThenChangePositionAsWell() throws Exception {
+        long start1 = 100;
+        long start2 = 200;
         SubSnpNoHgvs input = new SubSnpNoHgvs(SS_ID, RS_ID, REFERENCE_ALLELE, ALTERNATE_ALLELE, ASSEMBLY, "", "",
-                                              UCSC_1, START, UCSC_1, START,
-                                              DbsnpVariantType.SNV, Orientation.FORWARD, Orientation.FORWARD,
-                                              Orientation.FORWARD, true, true, true, true, null, null, TAXONOMY);
-        assertEquals(SEQNAME_1, processor.process(input).getContigName());
+                                              CHROMOSOME, start1, MISSING_CONTIG, start2, DbsnpVariantType.SNV,
+                                              Orientation.FORWARD, Orientation.FORWARD, Orientation.FORWARD, true, true,
+                                              true, true, null, null, TAXONOMY);
+        assertEquals(start1, processor.process(input).getContigStart());
```

Comment:
Can we check that the contig name is also the appropriate one, for the sake of completeness?